packages/core-mobile/ios/NotificationServiceExtensionInternal/NotificationService.swift

```diff
@@ -0,0 +1,37 @@
+//
+//  NotificationService.swift
+//  NotificationServiceExtensionInternal
+//
+//  Created by Ruijia Lin on 9/11/24.
+//
+
+import UserNotifications
+import UIKit
+
+class NotificationService: UNNotificationServiceExtension {
+
+    var contentHandler: ((UNNotificationContent) -> Void)?
+    var bestAttemptContent: UNMutableNotificationContent?
+    let defaults = UserDefaults(suiteName: "group.org.avalabs.corewallet")
+
+    override func didReceive(_ request: UNNotificationRequest, withContentHandler contentHandler: @escaping (UNNotificationContent) -> Void) {
+        self.contentHandler = contentHandler
+        bestAttemptContent = (request.content.mutableCopy() as? UNMutableNotificationContent)
+      var count: Int = defaults?.value(forKey: "badgeCount") as? Int ?? 0
+        if let bestAttemptContent = bestAttemptContent {
+            count = count + 1
+            defaults?.set(count, forKey: "badgeCount")
+            bestAttemptContent.badge = count as NSNumber
+            contentHandler(bestAttemptContent)
+        }
+    }
+    
+    override func serviceExtensionTimeWillExpire() {
+        // Called just before the extension will be terminated by the system.
+        // Use this as an opportunity to deliver your "best attempt" at modified content, otherwise the original push payload will be used.
+        if let contentHandler = contentHandler, let bestAttemptContent =  bestAttemptContent {
+            contentHandler(bestAttemptContent)
+        }
+    }
+}
+
```

Comment:
if we can have the same `NotificationService.swift`, I think we can just have a single file with multiple target membership.
<img width="469" alt="Screenshot 2024-09-16 at 11 06 54 AM" src="https://github.com/user-attachments/assets/57b34c17-9096-4a88-bacd-a8802ac83d3f">

Reply:
and currently, `AvaxWalletInternal` is also selected, but it seems unnecessary.

<img width="469" alt="Screenshot 2024-09-16 at 11 09 40 AM" src="https://github.com/user-attachments/assets/99f29b9f-75b4-4eb7-b4f7-229b375ecba6">

Reply:
[updated](238952ac669a47143fc5953850670c165d139c21)